documentation/howto/cdk_howto.adoc

```diff
@@ -0,0 +1,157 @@
+.Define a New CDK Server Adapter
+image::cdk_new_server_adapter.png[Define a New CDK Server Adapter]
++
+. Run the Container Development Environment server.
+. Add the OpenShift server.
+. Test the OpenShift Console at https://10.1.2.2:8443 with the username **openshift-dev** and password **devel**.
+
+You can now choose to continue working with OpenShift within JBoss Developer Studio or view instructions for Container-Based Development with JBoss Developer Studio.
+
+[[docker_tool]]
+== Docker Tooling
+
+After starting the CDK server in the IDE, you can follow one of two container development workflows:
+
+=== Using the Docker Tooling
+
+Use Docker for Container-Based Development as follows:
+
+. Create a new project with your **Dockerfile**.
+.. Click **File** > **New** and select **Project**.
+.. Type **java** in the search bar and from the results, select **Java Project** and click **Next** to continue.
+.. Add a name for the new project and click **Finish**.
+.. Click **File** > **New** and select **File**.
+.. Enter the **Dockerfile** as the name and click **Finish**.
+.. Edit the **Dockerfile** as desired and then save (for example, by creating a new Docker image to customize a given version of JBoss/WildFly by adding a datasource definition and its associated driver). The **Dockerfile** may also package your application as a **war** file via a Maven command, and copy it into the container in the WildFly deployments directory. See https://docs.docker.com/engine/reference/builder for more information about the **Dockerfile** instructions.
+. Do a Docker build using the Container Development Environment:
+.. In the **Project Explorer**, right-click the **Dockerfile** and select **Run As** > **Docker Image Build**.
+.. In the dialog box, for the **Connection** field, select your Container Development Environment server adapter.
+.. For **Repository Name**, enter the desired name for the docker image and click **OK**.
+.. Once the build is done, a new image with the given name will be listed in the **Docker Explorer** view and in the **Docker Images** view.
+. Do a Docker run using the Container Development Environment:
+.. Open the **Docker Explorer** view using the quick access menu using **Ctrl+3**.
+.. Navigate to the **Images** node under the Docker connection.
+.. Right-click your image and click **Run**.
+.. Fill in the necessary details and click **Finish** to run your image.  It is optional to give the container a name, but it is recommended to name it. This name helps locate the specific container in a list of containers in the future.
+. In the **Docker Explorer** view, select the container and expand its node and select the **8080** port and click on **Show In**>**Web browser** to access the application deployed in the Docker container.
+
+=== Next Steps for the Docker Tooling
+
+For further information about the basics of Docker Tooling, see link:docker_basics.html[Configure Docker Tooling (Basic)]
+
+=== Using OpenShift Tooling
```

Comment:
I would add a level 2 section title here, such as: "OpenShift Tooling" to better distinguish with the Docker tooling described above.